float8_experimental/float8_dynamic_linear.py

```diff
@@ -62,26 +62,26 @@ class Float8DynamicLinear(torch.nn.Linear):
     conversion to fp8 of the input and weight tensors.
     """
 
-    def __init__(self, use_activation_hooks: bool, **super_kwargs):
+    def __init__(self, cast_activation: bool, **super_kwargs):
         """
         Args:
-            use_activation_hooks (bool): whether to use activation hooks for casting to and from float8
+            cast_activation (bool): whether to do activation casting to and from float8
         """
         super().__init__(**super_kwargs)
 
-        self.use_activation_hooks = use_activation_hooks
+        self.cast_activation = cast_activation
 
     def forward(self, x):
         # cast x to float8_e4m3fn if not using activation hooks
-        x_fp8 = x if self.use_activation_hooks else self.cast_to_float8_e4m3fn(x)
+        x_fp8 = self.cast_to_float8_e4m3fn(x) if self.cast_activation else x
 
         # cast w to float8_e4m3fn
         w_fp8 = self.cast_to_float8_e4m3fn(self.weight)
 
         y = torch.nn.functional.linear(x_fp8, w_fp8, self.bias)
 
         # Cast gradY to float8_e5m2 during backward if not using activation hooks
-        if not self.use_activation_hooks:
+        if self.cast_activation:
```

Comment:
DTensor also can handle this?

Reply:
yeah both activation casting before `F.linear` and after `F.linear` now happens in TP hooks directly in https://github.com/pytorch-labs/float8_experimental/pull/234